Huge Refactor

declandewet opened the pull request:
This is not quite ready for merge yet, but I want to have it open so I can get feedback. What do you think so far, @jenius?

declandewet commented:
Okay, so the aliasing is causing some tests to fail - I'm not exactly sure what's happening, but if I change
`density` to `rupture-density` and then alias `if not rupture.no-conflict then density = rupture-density` (and subsequently for all friends), the `density` variable inside a few of the mixins gets overwritten to the value of `rupture-retina`. Stylus is just getting more and more weird.

jenius commented:
Looks good to me so far. Might as well squash those two commits together since they are doing the same thing as well.

declandewet commented:
As soon as I figure out this `no-conflict` issue :+1:

declandewet commented:
hey @jenius I'm having some trouble here that perhaps you can take a look at (I've already squashed the two commits you asked me to) - I'm trying to implement a `rupture-no-conflict` flag that when set to true, forces rupture to not define any variables without the `rupture-` prefix - but the way I've changed things around is causing some marginal errors and I'm not so sure I can tackle this without better opinion

jenius commented:
What are the errors it's causing? The code in the WIP commit looks pretty straightforward to me...

declandewet commented:
That's the thing - it's kind of tough to figure that out:

![](http://cl.ly/image/2m120m2H3w0c)

I was hoping you could checkout into this branch and see for yourself?

jenius commented:
Ah that's weird. Have you done the standard logging of `rupture-retina` and such? I could but it might be a bit as I have some other things I need to clean up.

declandewet commented:
I've gone through all those debugging steps and it just gets weirder and weirder. Pretty sure you know the feeling after all the sourcemap stuff you're doing for Roots, lol. Take your time, Rupture has basically neared it's final feature set, the only thing really missing is height based queries and this refactor - so there's no rush.

BerkeleyTrue commented:
Is this PR still in the works? It's been over a year and a no-conflict mode is badly needed.

jescalan commented:
@BerkeleyTrue no it isn't, but you are welcome to pick it up if you're interested!

W0LFMAN commented:
As i see [this issue](https://github.com/jescalan/rupture/issues/80) is resolved in this PR. Please when you merge this PR, close the issue.
